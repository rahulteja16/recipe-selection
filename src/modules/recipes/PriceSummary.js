import Box from '../../components/Box';
import Flex from '../../components/Flex';
import React from 'react';
import Text from '../../components/Text';
import { parseRawPrice } from './price';

const SummaryList = ({summary,shippingPrice,totalPrice }) => {
  return <Box width={['290px', '450px']} padding="md">
  {summary.map((item,index) =>
  (
    <Box key={index} data-testid="price-summary-item">
      <Flex justifyContent='space-between'>
        <Text fontSize="sm" fontWeight="regular">{item.name}  { item.qty !== 1 &&  ` x ${item.qty}`}</Text>
        <Text fontSize="sm" fontWeight="regular">{parseRawPrice(item.amount)}</Text>
      </Flex>
      <Box paddingTop="xs"></Box>
    </Box>
  )
  )}
  {shippingPrice !== 0 ?
    <Flex justifyContent='space-between' data-testid="price-summary-shipping" >
      <Text fontSize="sm" fontWeight="regular">Shipping</Text>
      <Text fontSize="sm" fontWeight="regular" data-testid="price-summary-shipping-item" >{shippingPrice}</Text>
    </Flex>
    : null}

  <Box paddingTop="xs"></Box>
  <Box borderTopWidth="sm" borderTopColor="border" borderTopStyle="solid" borderColor='#E4E4E4'></Box>
  <Box paddingTop="xs"></Box>
  <Flex justifyContent='space-between' data-testid="price-summary-total">
    <Text fontSize="sm" fontWeight="bold">Total</Text>
    <Text fontSize="sm" fontWeight="bold" data-testid="price-summary-total-item">{totalPrice}</Text>
  </Flex>
</Box>
}

const EmptySummary  = () => {
  return <Box width={['290px', '450px']} padding={16}>
    <Flex justifyContent='center' data-testid="price-summary-empty">
      <Text fontSize="sm" fontWeight="bold">Cart is Empty!!</Text>
    </Flex>
  </Box>
}


// Create PriceSummary user interface
const PriceSummary = ({ summary, totalPrice, shippingPrice }) => {

  if(summary.length !== 0) {
    return <SummaryList summary = {summary} totalPrice = {totalPrice} shippingPrice = {shippingPrice}/>;
  } else {
    return <EmptySummary />;
  }
};

export default PriceSummary;
