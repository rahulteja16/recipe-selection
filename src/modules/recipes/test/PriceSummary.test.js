import { render, screen } from'@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import PriceSummary from '../PriceSummary';
import React from 'react';

const props = {
  summary: [
    {id: "5f4d4a7e62fb0224951e7ec4", name: "Chicken Sausage & Spinach Ravioli", amount: 1798, qty: 1},
    {id: "5f4d4aa9f4508b34e9680613", name: "Gouda Vibes Burgers", amount: 1798, qty: 1},
    {id: "5f4d4acdab96be0cd6073022", name: "Figgy Balsamic Pork", amount: 1798, qty: 1}
  ],
  totalPrice: '$66.92',
  shippingPrice: '$12.98',
}

describe('Price Summary Component', ()=> {
  test('it renders the recipe item div',  () =>  {
      render(<PriceSummary {...props} />);
      expect(screen.getAllByTestId('price-summary-item')).toHaveLength(3);
  });
  test('it should render shipping price div', ()=> {
    render(<PriceSummary {...props} />);
    expect(screen.getAllByTestId('price-summary-shipping')).toHaveLength(1);
  });
  test('it should render shipping price item', ()=> {
    render(<PriceSummary {...props} />);
    expect(screen.getByTestId('price-summary-shipping-item')).toHaveTextContent(
      "$12.98"
    );;
  });
  test('it should render total price div', ()=> {
    render(<PriceSummary {...props} />);
    expect(screen.getAllByTestId('price-summary-total')).toHaveLength(1);
  });

  test('it should render total price item', ()=> {
    render(<PriceSummary {...props} />);
    expect(screen.getByTestId('price-summary-total-item')).toHaveTextContent(
      "$66.92"
    );;
  });

  test('it should render empty cart message, if no items in the cart', () => {
    const emptyProps = {...props};
    emptyProps.summary = [];
    render(<PriceSummary {...emptyProps} />);
    expect(screen.getAllByTestId('price-summary-empty')).toHaveLength(1);
  });
  test('it should not render shipping price, if shipping is 0', () => {
    const noShippingProps = {...props};
    noShippingProps.shippingPrice = 0;
    render(<PriceSummary {...noShippingProps} />);
    expect(screen.queryByTestId('price-summary-shipping')).not.toBeInTheDocument();
  });
  test('it should have a correct summary length', () => {
    render(<PriceSummary {...props} />);
    expect(screen.queryAllByTestId('price-summary-item')).toHaveLength(3);
  })
  test('it should match snapshot', ()=> {
    expect(render(<PriceSummary {...props} />)).toMatchSnapshot();
  });
  test('it should match empty state snapshot', () => {
    const emptyProps = {...props};
    emptyProps.summary = [];
    expect(render(<PriceSummary {...emptyProps} />)).toMatchSnapshot();
  });
});
