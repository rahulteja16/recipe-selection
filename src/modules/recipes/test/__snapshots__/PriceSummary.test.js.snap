// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Summary Component it should match empty state snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM elkMxd"
        width="290px,450px"
      >
        <div
          class="sc-bdnxRM sc-gtsrHT fcOtpP"
          data-testid="price-summary-empty"
          display="flex"
        >
          <div
            class="sc-dlnjwi gCGEFP"
            font-size="sm"
            font-weight="bold"
          >
            Cart is Empty!!
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM elkMxd"
      width="290px,450px"
    >
      <div
        class="sc-bdnxRM sc-gtsrHT fcOtpP"
        data-testid="price-summary-empty"
        display="flex"
      >
        <div
          class="sc-dlnjwi gCGEFP"
          font-size="sm"
          font-weight="bold"
        >
          Cart is Empty!!
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Price Summary Component it should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM jcpITf"
        width="290px,450px"
      >
        <div
          class="sc-bdnxRM lmKwmz"
          data-testid="price-summary-item"
        >
          <div
            class="sc-bdnxRM sc-gtsrHT iOuxLz"
            display="flex"
          >
            <div
              class="sc-dlnjwi cKxMDg"
              font-size="sm"
              font-weight="regular"
            >
              Chicken Sausage & Spinach Ravioli
                
            </div>
            <div
              class="sc-dlnjwi cKxMDg"
              font-size="sm"
              font-weight="regular"
            >
              $17.98
            </div>
          </div>
          <div
            class="sc-bdnxRM jhkwhO"
          />
        </div>
        <div
          class="sc-bdnxRM lmKwmz"
          data-testid="price-summary-item"
        >
          <div
            class="sc-bdnxRM sc-gtsrHT iOuxLz"
            display="flex"
          >
            <div
              class="sc-dlnjwi cKxMDg"
              font-size="sm"
              font-weight="regular"
            >
              Gouda Vibes Burgers
                
            </div>
            <div
              class="sc-dlnjwi cKxMDg"
              font-size="sm"
              font-weight="regular"
            >
              $17.98
            </div>
          </div>
          <div
            class="sc-bdnxRM jhkwhO"
          />
        </div>
        <div
          class="sc-bdnxRM lmKwmz"
          data-testid="price-summary-item"
        >
          <div
            class="sc-bdnxRM sc-gtsrHT iOuxLz"
            display="flex"
          >
            <div
              class="sc-dlnjwi cKxMDg"
              font-size="sm"
              font-weight="regular"
            >
              Figgy Balsamic Pork
                
            </div>
            <div
              class="sc-dlnjwi cKxMDg"
              font-size="sm"
              font-weight="regular"
            >
              $17.98
            </div>
          </div>
          <div
            class="sc-bdnxRM jhkwhO"
          />
        </div>
        <div
          class="sc-bdnxRM sc-gtsrHT iOuxLz"
          data-testid="price-summary-shipping"
          display="flex"
        >
          <div
            class="sc-dlnjwi cKxMDg"
            font-size="sm"
            font-weight="regular"
          >
            Shipping
          </div>
          <div
            class="sc-dlnjwi cKxMDg"
            data-testid="price-summary-shipping-item"
            font-size="sm"
            font-weight="regular"
          >
            $12.98
          </div>
        </div>
        <div
          class="sc-bdnxRM jhkwhO"
        />
        <div
          class="sc-bdnxRM kvgPit"
        />
        <div
          class="sc-bdnxRM jhkwhO"
        />
        <div
          class="sc-bdnxRM sc-gtsrHT iOuxLz"
          data-testid="price-summary-total"
          display="flex"
        >
          <div
            class="sc-dlnjwi gCGEFP"
            font-size="sm"
            font-weight="bold"
          >
            Total
          </div>
          <div
            class="sc-dlnjwi gCGEFP"
            data-testid="price-summary-total-item"
            font-size="sm"
            font-weight="bold"
          >
            $66.92
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM jcpITf"
      width="290px,450px"
    >
      <div
        class="sc-bdnxRM lmKwmz"
        data-testid="price-summary-item"
      >
        <div
          class="sc-bdnxRM sc-gtsrHT iOuxLz"
          display="flex"
        >
          <div
            class="sc-dlnjwi cKxMDg"
            font-size="sm"
            font-weight="regular"
          >
            Chicken Sausage & Spinach Ravioli
              
          </div>
          <div
            class="sc-dlnjwi cKxMDg"
            font-size="sm"
            font-weight="regular"
          >
            $17.98
          </div>
        </div>
        <div
          class="sc-bdnxRM jhkwhO"
        />
      </div>
      <div
        class="sc-bdnxRM lmKwmz"
        data-testid="price-summary-item"
      >
        <div
          class="sc-bdnxRM sc-gtsrHT iOuxLz"
          display="flex"
        >
          <div
            class="sc-dlnjwi cKxMDg"
            font-size="sm"
            font-weight="regular"
          >
            Gouda Vibes Burgers
              
          </div>
          <div
            class="sc-dlnjwi cKxMDg"
            font-size="sm"
            font-weight="regular"
          >
            $17.98
          </div>
        </div>
        <div
          class="sc-bdnxRM jhkwhO"
        />
      </div>
      <div
        class="sc-bdnxRM lmKwmz"
        data-testid="price-summary-item"
      >
        <div
          class="sc-bdnxRM sc-gtsrHT iOuxLz"
          display="flex"
        >
          <div
            class="sc-dlnjwi cKxMDg"
            font-size="sm"
            font-weight="regular"
          >
            Figgy Balsamic Pork
              
          </div>
          <div
            class="sc-dlnjwi cKxMDg"
            font-size="sm"
            font-weight="regular"
          >
            $17.98
          </div>
        </div>
        <div
          class="sc-bdnxRM jhkwhO"
        />
      </div>
      <div
        class="sc-bdnxRM sc-gtsrHT iOuxLz"
        data-testid="price-summary-shipping"
        display="flex"
      >
        <div
          class="sc-dlnjwi cKxMDg"
          font-size="sm"
          font-weight="regular"
        >
          Shipping
        </div>
        <div
          class="sc-dlnjwi cKxMDg"
          data-testid="price-summary-shipping-item"
          font-size="sm"
          font-weight="regular"
        >
          $12.98
        </div>
      </div>
      <div
        class="sc-bdnxRM jhkwhO"
      />
      <div
        class="sc-bdnxRM kvgPit"
      />
      <div
        class="sc-bdnxRM jhkwhO"
      />
      <div
        class="sc-bdnxRM sc-gtsrHT iOuxLz"
        data-testid="price-summary-total"
        display="flex"
      >
        <div
          class="sc-dlnjwi gCGEFP"
          font-size="sm"
          font-weight="bold"
        >
          Total
        </div>
        <div
          class="sc-dlnjwi gCGEFP"
          data-testid="price-summary-total-item"
          font-size="sm"
          font-weight="bold"
        >
          $66.92
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
